x-pack/legacy/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -248,79 +323,54 @@ export function XYChart({ data, args, formatFactory, timeZone, chartTheme }: XYC
           }
 
           const columnToLabelMap = columnToLabel ? JSON.parse(columnToLabel) : {};
-          const splitAccessorLabel = splitAccessor ? columnToLabelMap[splitAccessor] : '';
-          const yAccessors = accessors.map(accessor => columnToLabelMap[accessor] || accessor);
-          const idForLegend = splitAccessorLabel || yAccessors;
-          const sanitized = sanitizeRows({
-            splitAccessor,
-            formatFactory,
-            columnToLabelMap,
-            table: data.tables[layerId],
+          const idForLegend = accessors;
+          const table = data.tables[layerId];
+
+          const rows = table.rows.filter(row => {
+            if (
+              splitAccessor &&
+              !row[splitAccessor] &&
+              accessors.every(accessor => !row[accessor])
+            ) {
+              return false;
+            }
+            return true;
```

Comment:
Nitpick: When the only options are `return true` or `return false`, you can return the condition itself:

```
return (...conditions...);
```